web/src/app/pages/create-survey/data-sharing-terms/data-sharing-terms.component.html

```diff
@@ -0,0 +1,55 @@
+<!--
+  Copyright 2024 The Ground Authors.
+
+  Licensed under the Apache License, Version 2.0 (the "License");
+  you may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+      https://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+-->
+
+<form [formGroup]="formGroup">
+  <mat-radio-group class="data-sharing-terms"
+    [id]="typeControlKey"
+    [formControlName]="typeControlKey"
+  >
+    <mat-card *ngFor="let option of dataSharingTermsOptions"
```

Comment:
Totally optional, but we can now use the `@for` and `@if` syntax in Angular since we are on a more recent version now and it can be more readable: https://angular.dev/api/core/@for.

Reply:
Cool, yeah, will probably do this together with updating to use the harness, in a follow-up.